### DISCUSSION TITLE
[Chapter 08] GenerationType, Propagation 속성

### DISCUSSION THREAD

**Discussion by BvrPark** [cowaine/spring-boot-for-msa#101] [Chapter 08. Data Persistence]
### 1. GenerationType.IDENTITY vs SEQUENCE
원래 JPA는 쿼리를 임시 저장소(?) 같은 곳에다가 모아뒀다가 강제 플러시나 트랜잭션 커밋 시점 등에 쿼리를 한번에 DB에 보내는 것으로 아는데

**GenerationType.IDENTITY**는 DB에 기본 키 생성을 맡기기 때문에 Insert 쿼리를 날려야지 ID값을 알 수 있어서 즉시 insert 쿼리를 날리는 것으로 알고 있고 **SEQUENCE**는 DB가 관리하는 객체에서 ID값을 가져와서 즉시 insert 쿼리를 날리지 않는다고 알고 있습니다.

이게 insert가 엔티티를 DB에 저장할 때 엔티티의 ID값(기본키)가 아직 생성이 됐을 수도 있고 아닐 수도 있어서 두 전략에 있어서 차이가 나타나는 것이고 다른 쿼리들 select, update, delete 등은 이미 엔티티의 ID값이 생성되어 있으니까 두 전략에 차이가 없는건가요?

### 2. Propagation 속성
- p.517 ~ p.518에서 Propagation 속성들 설명 중에 몇몇은 트랜잭션 없이 쿼리를 진행한다고 나와있던데 그러면 작업이 실패했을 때, 롤백없이 쭉 진행된다는 말인가요?
- 그렇다면 제 생각엔 무조건 트랜잭션이 있는 상태에서 쿼리를 진행하는 것이 좋은 것 같은데 왜 저런 속성이 생겼을까요? 트랜잭션 없이 쓰이는 상황이 어떤 경우가 있나요?

**Comment by ahngilwoong** [comment]
Q1. 저도 말씀 그대로가 맞다고 생각합니다.

Q2. 트랜잭션도 결국 비용이 생기긴 할텐데
페이지에서 자주 사용되는(데이터 폴링 등?) 무거운 조회 작업에서 뭔가 쪼금이라도 어떻게든 비용을 줄이고자 할 때 ? 안쓰지 않을까 생각도 개인적으로 해봅니다 !

**Comment by Irisation23** [comment]
### 금주 간단정리 Check List

1. `2.`의 질문은 왜 `@Tranctional`을 달아놓고서 트랜잭션 없이 쿼리를 진행한다고 설명을 했는지에 대한 질문임. (e.g SUPPORT)
2. 답변 중 예를 들어 로그를 기록할 때 `출금`에 관련된 로그는 성공 실패 여부에 상관이 없기 때문에 기록을 궂이 안해도 됨.
